src/Symfony/Component/HttpKernel/Event/FilterControllerArgumentsEvent.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Event;
+
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Allows filtering of controller arguments.
+ *
+ * You can call getController() to retrieve the controller and getArguments
+ * to retrieve the current arguments. With setArguments() you can replace
+ * arguments that are used to call the controller.
+ *
+ * Arguments set in the event must be compatible with the signature of the
+ * controller.
+ *
+ * @author Christophe Coevoet <stof@notk.org>
+ */
+class FilterControllerArgumentsEvent extends KernelEvent
+{
+    private $controller;
+    private $arguments;
+
+    public function __construct(HttpKernelInterface $kernel, callable $controller, array $arguments, Request $request, $requestType)
+    {
+        parent::__construct($kernel, $request, $requestType);
+
+        $this->controller = $controller;
+        $this->arguments = $arguments;
+    }
+
+    /**
+     * Returns the current controller.
+     *
+     * @return callable
+     */
+    public function getController()
+    {
+        return $this->controller;
+    }
+
+    /**
+     * @return array
+     */
+    public function getArguments()
+    {
+        return $this->arguments;
+    }
+
+    /**
+     * @param array $arguments
+     */
+    public function setArguments(array $arguments)
+    {
+        $this->arguments = $arguments;
+    }
```

Comment:
Yes, you can implement your own `ControllerResolver`, but to add your own notation while supporting the existing ones, you'd have to extend the existing classes + service definitions etc, which is quite cumbersome. Anyways, this is getting off-topic. 

I really don't mind if you keep the `setArguments()` as you have a good reason to add it: Another extension point which is easy to implement. I'm just afraid it will get confusing if you have 2 extension points.

Would it be possible to log it if arguments have changed (even if only in dev)?